### ISSUE TITLE
Check DOI Resolution against sample set

### ISSUE THREAD

**Issue by BenJam** [mysociety/remit#136]
My own sample of one isn't enough but:

DOI ref: 10.1056/NEJMoa1506110
link: http://www.nejm.org/doi/10.1056/NEJMoa1506110
checks out okay on http://www.crossref.org/05researchers/58doi_resolver.html 

See #6

**Comment by stevenday** [comment]
The root of this is that something about the response crashes the BibURI library I'm using:

```
1.9.3-p551 :002 > BibURI.lookup('doi:10.1056/NEJMoa1506110')
NoMethodError: undefined method `+' for nil:NilClass
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/biburi-0.1.1/lib/biburi/driver/coins.rb:194:in `coins_to_bibtex'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/biburi-0.1.1/lib/biburi/driver/doi.rb:83:in `lookup'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/biburi-0.1.1/lib/biburi.rb:22:in `block in lookup'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/biburi-0.1.1/lib/biburi.rb:20:in `each'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/biburi-0.1.1/lib/biburi.rb:20:in `lookup'
	from (irb):2
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/railties-4.2.5.1/lib/rails/commands/console.rb:110:in `start'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/railties-4.2.5.1/lib/rails/commands/console.rb:9:in `start'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/railties-4.2.5.1/lib/rails/commands/commands_tasks.rb:68:in `console'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/railties-4.2.5.1/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/railties-4.2.5.1/lib/rails/commands.rb:17:in `<top (required)>'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/activesupport-4.2.5.1/lib/active_support/dependencies.rb:274:in `require'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/activesupport-4.2.5.1/lib/active_support/dependencies.rb:274:in `block in require'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/activesupport-4.2.5.1/lib/active_support/dependencies.rb:240:in `load_dependency'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/activesupport-4.2.5.1/lib/active_support/dependencies.rb:274:in `require'
	from /vagrant/bin/rails:9:in `<top (required)>'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/activesupport-4.2.5.1/lib/active_support/dependencies.rb:268:in `load'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/activesupport-4.2.5.1/lib/active_support/dependencies.rb:268:in `block in load'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/activesupport-4.2.5.1/lib/active_support/dependencies.rb:240:in `load_dependency'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/activesupport-4.2.5.1/lib/active_support/dependencies.rb:268:in `load'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/spring-1.6.0/lib/spring/commands/rails.rb:6:in `call'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/spring-1.6.0/lib/spring/command_wrapper.rb:38:in `call'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/spring-1.6.0/lib/spring/application.rb:183:in `block in serve'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/spring-1.6.0/lib/spring/application.rb:156:in `fork'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/spring-1.6.0/lib/spring/application.rb:156:in `serve'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/spring-1.6.0/lib/spring/application.rb:131:in `block in run'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/spring-1.6.0/lib/spring/application.rb:125:in `loop'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/spring-1.6.0/lib/spring/application.rb:125:in `run'
	from /home/vagrant/.rvm/gems/ruby-1.9.3-p551/gems/spring-1.6.0/lib/spring/application/boot.rb:18:in `<top (required)>'
	from /home/vagrant/.rvm/rubies/ruby-1.9.3-p551/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /home/vagrant/.rvm/rubies/ruby-1.9.3-p551/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:54:in `require'
```

**Comment by BenJam** [comment]
This seems to happen with other references from the same publication. Need a realistic sample.  Assigning to @kimmwest to get some.

**Comment by stevenday** [comment]
I've stopped this crashing on DOI's it has a problem with, but we still need to figure out *why* it crashes.

**Comment by stevenday** [comment]
I've fixed this issue in 7c6daf0 so that I can't seem to find a DOI which breaks it.

I'm going to close this issue so that we just have #6 for @kimmwest to comment on.
